Source/DafnyCore/DafnyOptions.cs

```diff
@@ -188,16 +188,79 @@ public class DafnyOptions : Bpl.CommandLineOptions {
 
       try {
         if (i >= arguments.Length) {
-          return base.Parse(arguments);
+          return BaseParse(arguments);
         }
         MainArgs = arguments.Skip(i + 1).ToList();
-        return base.Parse(arguments.Take(i).ToArray());
+        return BaseParse(arguments.Take(i).ToArray());
       } catch (Exception e) {
         ErrorWriter.WriteLine("Invalid filename: " + e.Message);
         return false;
       }
     }
 
+    protected override Bpl.CommandLineParseState InitializeCommandLineParseState(string[] args) {
+      return new TextWriterParseState(args, ToolName, ErrorWriter);
+    }
+
+    /// <summary>
+    /// Needed because the Boogie version writes to Console.Error
+    /// </summary>
+    class TextWriterParseState : Bpl.CommandLineParseState {
+      private readonly TextWriter errorWriter;
+
+      public TextWriterParseState(string[] args, string toolName, TextWriter errorWriter) : base(args, toolName) {
+        this.errorWriter = errorWriter;
+      }
+
+      public override void Error(string message, params string[] args) {
+        errorWriter.WriteLine("{0}: Error: {1}", ToolName, string.Format(message, args));
+        EncounteredErrors = true;
+      }
+    }
+
+    /// <summary>
+    /// Customized version of Microsoft.Boogie.CommandLineOptions.Parse
+    /// Needed because the Boogie version writes to Console.Error
+    /// </summary>
+    private bool BaseParse(string[] args) {
+      Environment = Environment + "Command Line Options: " + string.Join(" ", args);
+      args = cce.NonNull<string[]>((string[])args.Clone());
+      Bpl.CommandLineParseState state;
+      for (state = InitializeCommandLineParseState(args); state.i < args.Length; state.i = state.nextIndex) {
+        cce.LoopInvariant(state.args == args);
+        string file = args[state.i];
+        state.s = file.Trim();
+        bool flag = state.s.StartsWith("-") || state.s.StartsWith("/");
+        int length = state.s.IndexOf(':');
+        if (0 <= length & flag) {
+          state.hasColonArgument = true;
+          args[state.i] = state.s.Substring(length + 1);
+          state.s = state.s.Substring(0, length);
+        } else {
+          ++state.i;
+          state.hasColonArgument = false;
+        }
+        state.nextIndex = state.i;
+        if (flag) {
+          if (!ParseOption(state.s.Substring(1), state)) {
+            if (Path.DirectorySeparatorChar == '/' && state.s.StartsWith("/")) {
+              AddFile(file, state);
+            } else {
+              UnknownSwitch(state);
+            }
+          }
+        } else {
+          AddFile(file, state);
+        }
+      }
+      if (state.EncounteredErrors) {
+        ErrorWriter.WriteLine("Use /help for available options");
```

Comment:
```suggestion
        ErrorWriter.WriteLine("Use --help for available options");
```
Since we can now customize this error message, I thought we could as well support the new CLI by default.

Reply:
I'm not sure. This is printed when you're already using the old CLI.